### PR TITLE
[CL Incentives] Uptime Accumulator Initialization

### PR DESCRIPTION
**Pull request by AlpinYukseloglu**
Closes: #4112

## What is the purpose of the change

This PR implements uptime accumulator initialization logic for CL incentives.

## Brief Changelog

- Add list of supported uptimes to `types/constants.go` (1 min, 1h, 1d, 1w)
- Upon pool creation, initialize an accumulator for each of these using the Accumulator API package

## Testing and Verifying

- New functionality is tested in `x/concentrated-liquidity/pool_test.go`

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (no)
  - How is the feature or change documented? (not documented)

### CODE REVIEW

**Review by czarcas7ic — COMMENTED** [review]
This LGTM but I think we should have gotests for `createUptimeAccumulators`, `getUptimeAccumulatorName`, and `GetUptimeAccumulators`

File: x/concentrated-liquidity/incentives.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+func getUptimeAccumulatorName(poolId uint64, uptimeId uint64) string {
+	poolIdStr := strconv.FormatUint(poolId, uintBase)
+	uptimeIdStr := strconv.FormatUint(uptimeId, uintBase)
+	return strings.Join([]string{uptimeAccumPrefix, uptimeIdStr, poolIdStr}, "/")
+}
+
+// nolint: unused
+// getUptimeAccumulators gets the uptime accumulator objects for the given poolId
+// Returns error if accumulator for the given poolId does not exist.
+func (k Keeper) GetUptimeAccumulators(ctx sdk.Context, poolId uint64) ([]accum.AccumulatorObject, error) {
```
Comment (p0mvn) [inline x/concentrated-liquidity/incentives.go:39]: Does this have to be exported?

File: x/concentrated-liquidity/pool_test.go [hunk 2]
```diff
@@ -85,6 +84,14 @@ func (s *KeeperTestSuite) TestInitializePool() {
 				feeAccumulator, err := s.App.ConcentratedLiquidityKeeper.GetFeeAccumulator(s.Ctx, test.poolI.GetId())
 				s.Require().NoError(err)
 				s.Require().Equal(sdk.DecCoins(nil), feeAccumulator.GetValue())
+
+				// Ensure that uptime accumulators have been properly initialized
+				uptimeAccumulators, err := s.App.ConcentratedLiquidityKeeper.GetUptimeAccumulators(s.Ctx, test.poolI.GetId())
+				s.Require().NoError(err)
+				s.Require().Equal(len(types.SupportedUptimes), len(uptimeAccumulators))
+				for _, uptimeAccumulator := range uptimeAccumulators {
+					s.Require().Equal(sdk.DecCoins(nil), uptimeAccumulator.GetValue())
```
Comment (p0mvn) [inline x/concentrated-liquidity/pool_test.go:93]:
```suggestion
					s.Require().Equal(cl.EmptyCoins, uptimeAccumulator.GetValue())
```

**Review by p0mvn — COMMENTED** [review]
lgtm after tests are added

File: x/concentrated-liquidity/incentives.go (hunk 1 above)
Comment (AlpinYukseloglu) [reply x/concentrated-liquidity/incentives.go:39]: I figured it could be useful & it's a getter which we tend to make public in GAMM

Comment (AlpinYukseloglu) [reply x/concentrated-liquidity/incentives.go]: Ended up making private and exporting

**Comment by AlpinYukseloglu** [comment]
All the behavior was already tested at the pool initialization level, but I added lower level sanity tests for any new functions for the sake of coverage

**Review by czarcas7ic — APPROVED** [review]
Thanks for adding the tests, LGTM!

File: x/concentrated-liquidity/incentives_test.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
+package concentrated_liquidity_test
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	cl "github.com/osmosis-labs/osmosis/v14/x/concentrated-liquidity"
+	"github.com/osmosis-labs/osmosis/v14/x/concentrated-liquidity/types"
+)
+
+var defaultPoolId = uint64(1)
```
Comment (p0mvn) [inline x/concentrated-liquidity/incentives_test.go:10]: nit: could be const

**Review by p0mvn — APPROVED** [review]
Nice

**Pull request merged**